SPHINXsys/src/for_3D_build/geometries/geometry.h

```diff
@@ -86,17 +93,32 @@ namespace SPH
 		Vec3d findClosestPoint(const Vec3d &input_pnt);
 
 	public:
-		ComplexShape() : Shape("ComplexShape"){};
-		ComplexShape(std::string complex_shape_name) : Shape(complex_shape_name){};
-		virtual ~ComplexShape(){};
+		ComplexShape() : Shape("ComplexShape") { complex_shape_mesh_ = nullptr; };
+		ComplexShape(std::string complex_shape_name) : Shape(complex_shape_name) { complex_shape_mesh_ = nullptr; };
+		ComplexShape(ComplexShapeMesh*complex_shape_mesh) : Shape("ComplexShape") { complex_shape_mesh_ = complex_shape_mesh; };
+		virtual ~ComplexShape() {};
 		virtual BoundingBox findBounds() override;
 
-		void addTriangleMeshShape(TriangleMeshShape *triangle_mesh_shape, ShapeBooleanOps op);
-		void addComplexShape(ComplexShape *complex_shape, ShapeBooleanOps op);
-		void addBrick(Vec3d halfsize, int resolution, Vec3d translation, ShapeBooleanOps op);
-		void addSphere(Real radius, int resolution, Vec3d translation, ShapeBooleanOps op);
-		void addCylinder(SimTK::UnitVec3 axis, Real radius, Real halflength, int resolution, Vec3d translation, ShapeBooleanOps op);
-		void addFormSTLFile(std::string file_path_name, Vec3d translation, Real scale_factor, ShapeBooleanOps op);
+		void addTriangleMeshShape(TriangleMeshShape *triangle_mesh_shape, ShapeBooleanOps op)		
+		{
+			if(complex_shape_mesh_->getName() == "ComplexShapeTriangleMesh")
+			{
+				ComplexShapeTriangleMesh *complex_shape_mesh = dynamic_cast<ComplexShapeTriangleMesh*>(complex_shape_mesh_);
+				complex_shape_mesh->addTriangleMeshShape(triangle_mesh_shape, op);
+			}
+		}
+		// void addComplexShape(ComplexShape *complex_shape, ShapeBooleanOps op);
+		void addBrick(Vec3d halfsize, int resolution, Vec3d translation, ShapeBooleanOps op)		
+		{
+			if(complex_shape_mesh_->getName() == "ComplexShapeTriangleMesh")
```

Comment:
please define "ComplexShapeTriangleMesh" as constant since it  has been used in several places.

Reply:
Those "ComplexShapeTriangleMesh" will be finally removed.
Define a function of addBrick() at level of complexShape Class is not necessary, it should be removed.
For now to minimize the affection on other test cases, I have kept this function.